Sources/ProjectDescription/SettingsTransformers.swift

```diff
@@ -0,0 +1,147 @@
+import Foundation
+
+extension Dictionary {
+    mutating func merge(with dict: [Key: Value]) {
+        dict.forEach { self[$0.key] = $0.value }
+    }
+
+    func merging(with dict: [Key: Value]) -> Dictionary {
+        var copy = self
+        copy.merge(with: dict)
+        return copy
+    }
+}
+
+private extension Bool {
+    var asSettingValue: SettingValue {
+        self ? "YES" : "NO"
+    }
+}
+
+private extension String {
+    var asSettingValue: SettingValue {
+        .init(stringLiteral: self)
+    }
+}
+
+public enum SwiftCompilationMode: String {
+    case singlefile
+    case wholemodule
+}
+
+public enum SwiftOptimizationLevel: String {
+    case o = "-O"
+    case oNone = "-Onone"
+    case oSize = "-Osize"
+}
+
+public extension SettingsDictionary {
+    // MARK: - Deployment target and SDKROOT
+
+    /// Sets "IPHONEOS_DEPLOYMENT_TARGET" to `value`
+    func iPhoneOSDeploymentTarget(_ value: String) -> SettingsDictionary {
+        merging(with: ["IPHONEOS_DEPLOYMENT_TARGET": value.asSettingValue])
+    }
+
+    /// Sets "SDKROOT" to "iphoneos"
+    func iPhoneOSasSDKRoot() -> SettingsDictionary {
+        merging(with: ["SDKROOT": "iphoneos"])
+    }
+
+    /// Sets "SDKROOT" to `value`
+    func sdkRoot(_ value: String) -> SettingsDictionary {
+        merging(with: ["SDKROOT": value.asSettingValue])
+    }
+
+    // MARK: - Code signing
+
+    /// Sets "CODE_SIGN_STYLE" to "Manual"
+    func manualCodeSigning() -> SettingsDictionary {
+        merging(with: ["CODE_SIGN_STYLE": "Manual"])
```

Comment:
Similarly to `automaticCodeSigning`, I'd let a developer specify `devTeam` - but for manual signing they will also need `PROVISIONING_PROFILE_SPECIFIER` and `CODE_SIGN_IDENTITY`.